src/fsharp/tast.fs

```diff
@@ -970,6 +970,17 @@ type Entity =
     /// Sets the structness of a record or union type definition
     member x.SetIsStructRecordOrUnion b  = let x = x.Data in let flags = x.entity_flags in x.entity_flags <- EntityFlags(flags.IsPrefixDisplay, flags.IsModuleOrNamespace, flags.PreEstablishedHasDefaultConstructor, flags.HasSelfReferentialConstructor, b)
 
+and [<RequireQualifiedAccess>] MaybeLazy<'T> =
+    | Strict of 'T
```

Comment:
This is a good change, thanks.  The name is a bit confusing since it might be confused with the Haskell terminology of ``maybe`` for  ``option``.  Also in future please make sure we have ``///`` comments for all new types, methods and properties for these core types.